src/osgUtil/RenderStage.cpp

```diff
@@ -511,22 +511,41 @@ void RenderStage::runCameraSetUp(osg::RenderInfo& renderInfo)
 
             fbo->apply(state);
 
-            // If no color attachment make sure to set glDrawBuffer/glReadBuffer to none
-            // otherwise glCheckFramebufferStatus will fail
-            // It has to be done after call to glBindFramebuffer (fbo->apply)
-            // and before call to glCheckFramebufferStatus
+#if !defined(OSG_GLES1_AVAILABLE) && !defined(OSG_GLES2_AVAILABLE) && !defined(OSG_GLES3_AVAILABLE)
             if ( !colorAttached )
             {
-            #if !defined(OSG_GLES1_AVAILABLE) && !defined(OSG_GLES2_AVAILABLE) && !defined(OSG_GLES3_AVAILABLE)
-                setDrawBuffer( GL_NONE, false );
-                setReadBuffer( GL_NONE, false );
+                // Coerce glCheckFramebufferStatus to accept FBOs without a color buffer.
+                // This is done by setting glDrawBuffer/glReadBuffer to GL_NONE.
+                // If this is not done, glCheckFramebufferStatus will fail with particular GL drivers (<= 4.1 ?)
+                // It has to be done after calling glBindFramebuffer (fbo->apply)
+                // and before calling glCheckFramebufferStatus.
+                // This can happen when an FBO camera attaches its color buffer later (dynamically).
+                // Notes:
+                // - To reach here a camera must call setImplicitBufferAttachmentMask() to disable implicit color buffer creation.
+                // - A camera can attach all its buffers later and have no buffers attached when reaching here.
+                // - GL_NONE is not an accepted value for glReadBuffer (?)
+                glPushAttrib( GL_COLOR_BUFFER_BIT );
```

Comment:
I don't have a GLES device to check addressed issue but AFAIK glPushAttrib / glPopAttrib are deprecated, ... i'm surprised it still work with GLES core context...

Reply:
The whole block is #ifdef'ed away when GLES is available. Was like this before my changes.
Not sure it answers your question though... I am not knowledgeable in GLES specificity.

Reply:
That's not a gles specificity, it's just the new guidelines...it's just to make sure your patch continue to work in the improbable case vendors remove deprecated funcs from drivers in the future

Reply:
Is it not enough to #ifdef that block away if GLES is available ?

Reply:
In a nutshell: the issue affects  cameras that render to an FBO and explicitly tell OSG to not create a color buffer. The color buffer will be attached later dynamically. osgEarth has a few such camera. OSG has none.

Reply:
I've no doubt you intention are well funded but as it concern very sensitive part of the sg management I would recommend to be more "community-explicit" and begin with a thread on general forum explaining the issue you want to fix and an example illustrating the usecase that doesn't involve osg hacking...